flavors/posture.go

```diff
@@ -174,18 +186,13 @@ func (bt *posture) Run(b *beat.Beat) error {
 	}
 }
 
-func initRegistry(log *logp.Logger, cfg *config.Config, ch chan fetching.ResourceInfo, le uniqueness.Manager) (fetchersManager.FetchersRegistry, error) {
-	registry := fetchersManager.NewFetcherRegistry(log)
-
-	parsedList, err := fetchersManager.Factories.ParseConfigFetchers(log, cfg, ch)
+func initRegistry(ctx context.Context, log *logp.Logger, cfg *config.Config, ch chan fetching.ResourceInfo, le uniqueness.Manager, k8sClient k8s.Interface, identityProvider func(cfg awssdk.Config) awslib.IdentityProviderGetter, awsConfigProvider awslib.ConfigProviderAPI) (registry.FetchersRegistry, error) {
+	f, err := factory.NewFactory(ctx, log, cfg, ch, le, k8sClient, identityProvider, awsConfigProvider)
 	if err != nil {
 		return nil, err
 	}
 
-	if err := registry.RegisterFetchers(parsedList, le); err != nil {
-		return nil, err
-	}
-
+	registry := registry.NewFetcherRegistry(log, f)
 	return registry, nil
```

Comment:
```suggestion
	return registry.NewFetcherRegistry(log, f), nil
```

Otherwise, variable 'registry' collides with imported package name